core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -221,7 +222,20 @@ void *CudaSpace::impl_allocate(
     const Kokkos::Tools::SpaceHandle arg_handle) const {
   void *ptr = nullptr;
 
+#ifndef CUDART_VERSION
+#error CUDART_VERSION undefined!
+#elif (CUDART_VERSION >= 11020)
+  cudaError_t error_code;
+  if ( (size_t)arg_alloc_size < (std::numeric_limits<size_t>::max() - 1000) ) {
+    error_code = cudaMallocAsync(&ptr, arg_alloc_size, 0);
+    cudaDeviceSynchronize();
+  }
+  else {
+    error_code = cudaErrorInvalidValue;
+  }
+#else
```

Comment:
Since this is a performance improvement, I would fall back to the original approach, i.e. using `cudaMalloc`, if the bug is triggered. Also, we should add a FIXME with an explanation so that we can remove the restriction eventually.

Reply:
I agree, I will make these changes now

Reply:
@masterleinad Thanks for your comments, I added these changes